nodejs/week2/meal-sharing/src/backend/api/reservation-router.js

```diff
@@ -0,0 +1,26 @@
+const express = require("express");
+const req = require("express/lib/request");
+const router = express.Router();
+
+const reservations = require("./../data/reservations.json");
+
+router.get("/", async (request, response) => {
+  response.send(reservations);
+});
+
+router.get("/:id", async (request, response) => {
+  const reservationId = Number(request.params.id);
+  if (!isNaN(reservationId) && reservationId) {
+    let reservation = {};
+    reservations.forEach((item) => {
```

Comment:
Instead of forEach you could use filter. something like:
const results = reservations.filter(reservation => reservation.id === reservationId)
...

Reply:
@jslambda I did in this way just because the test that was prepared for this homework faild in case I have array with a singe object inside. This test expects singe object.

Reply:
You can extract elements from an array: results[0]
but In this case reservations.find(reservation => reservation.id === reservationId) makes more sense.